### PR TITLE
Upgrade page scaffold

### PR DESCRIPTION
**Pull request by theswerd**
**Please follow this checklist. Please check each appropriate box (put an 'x' or check it after creating the PR).**
- [ ] REQUIRED: Do you have an Issue **assigned to you** within the v1 milestone for this PR?  Put the Issue number here:
- [x] Provided detailed pull request description and a succinct title (consider template below for guidance).
- [x] Tested your changes, especially after any code review iterations.
- [x] Included any relevant screenshots of UI updates.
- [x] Followed the [Contributor Guidelines](https://github.com/WorldHealthOrganization/app/blob/master/docs/CONTRIBUTING.md).
- [x] Verified all contributions are properly licensed pursuant to the [LICENSE](https://github.com/WorldHealthOrganization/app/blob/master/LICENSE) file in the root of the repository.
- [x] Verified your name is in the [content/credits.yaml](https://github.com/WorldHealthOrganization/app/blob/master/content/credits.yaml) file (if you want it to be).

After all boxes above are checked, request and receive an Approved review from any team member knowledgable in the area (TODO team member list).  Once approved, the team member will assign your review to a Committer or use the `needs-merge` label.

## What does this PR accomplish?

- gets rid of the listOfItems.dart file because it just does what PageScaffold does with slightly different styling

- extracts shareBar class

- creates page_scaffold folder for all components that go into page_scaffold

- Added option for sharebar at the bottom to pageScaffold
### Extra info
The only intentional difference I could find between list of items and PageScaffold was that listOfItems had the shareBar at the bottom, so I extracted it as a widget, and added an optional parameter to PageScaffold so it can be added to any page with a boolean

## Did you add any dependencies?

No

## How did you test the change?

iOS simulator

Simulator Screen Shot - iPhone 8 Plus - 2020-03-30 at 22 42 53
Simulator Screen Shot - iPhone 8 Plus - 2020-03-30 at 22 43 05

### CODE REVIEW

File: client/flutter/lib/components/emojiHeader.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class EmojiHeader extends StatelessWidget {
+  EmojiHeader(this.emoji);
+
+  final String emoji;
+
+  @override
+  Widget build(BuildContext context) {
+    return Center(
+      child: Text(
+        this.emoji,
```
Comment (PieterAelse) [inline client/flutter/lib/components/emojiHeader.dart]: `this` is not needed

File: client/flutter/lib/components/page_scaffold/page_scaffold.dart [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  final BuildContext context;
+  final EdgeInsets padding;
+  final bool showBackButton;
+  final bool showShareBottomBar;
+
+  PageScaffold(this.context,
+      {@required this.body,
+      @required this.title,
+       this.showShareBottomBar = false,
+      this.subtitle = "WHO COVID-19 App",
```
Comment (advayDev1) [inline client/flutter/lib/components/page_scaffold/page_scaffold.dart]:
```suggestion
      this.subtitle = "COVID-19 App",
```

File: client/flutter/lib/pages/travel_advice.dart [hunk 3]
```diff
@@ -8,6 +8,7 @@ class TravelAdvice extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return PageScaffold(context,
+    showShareBottomBar: true,
```
Comment (advayDev1) [inline client/flutter/lib/pages/travel_advice.dart]:
```suggestion
    showShareBottomBar: false,
```

File: client/flutter/pubspec.lock [hunk 4]
```diff
@@ -63,7 +63,7 @@ packages:
       name: csslib
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "0.16.1"
+    version: "0.16.0"
```
Comment (advayDev1) [inline client/flutter/pubspec.lock]: why would we go backwards?

File: client/flutter/lib/components/page_scaffold/page_scaffold.dart [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import './share_bar.dart';
+import 'package:flutter/material.dart';
+
+class PageScaffold extends StatelessWidget {
+  final String title;
+  final String subtitle;
+
+  final List<Widget> body;
+
+  final BuildContext context;
```
Comment (PieterAelse) [inline client/flutter/lib/components/page_scaffold/page_scaffold.dart]: This `context` is not used. The `PageHeader` in the build method used the `context` from its params. Which is working fine.
So this `context` should be removed, as it's also not a good practice to pass contexts to widgets in constructors (they get it through the build method anyway)

File: client/flutter/lib/components/page_scaffold/share_bar.dart [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import 'package:WHOFlutter/generated/l10n.dart';
+import 'package:flutter/material.dart';
+import 'package:share/share.dart';
+
```
Comment (PieterAelse) [inline client/flutter/lib/components/page_scaffold/share_bar.dart]: Empty line can be removed

File: client/flutter/pubspec.lock (hunk 4 above)
Comment (theswerd) [reply client/flutter/pubspec.lock]: Fixed

File: client/flutter/lib/pages/protect_yourself.dart [hunk 7]
```diff
@@ -103,38 +100,46 @@ class ProtectYourself extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     var localized = S.of(context);
```
Comment (PieterAelse) [inline client/flutter/lib/pages/protect_yourself.dart]: Should be `final`

File: client/flutter/lib/components/emojiHeader.dart (hunk 1 above)
Comment (theswerd) [reply client/flutter/lib/components/emojiHeader.dart]: Its good practice to be clear

File: client/flutter/lib/pages/protect_yourself.dart [hunk 8]
```diff
@@ -103,38 +100,46 @@ class ProtectYourself extends StatelessWidget {
...
-        ProtectYourselfCard(
-          message: _message(localized.protectYourselfListOfItemsPageListItem5),
-          child: _maskImage,
-        ),
-      ],
-      title: localized.protectYourselfTitle,
-    );
+    return PageScaffold(context,
+        title: S.of(context).protectYourselfTitle,
+        showShareBottomBar: true,
```
Comment (advayDev1) [inline client/flutter/lib/pages/protect_yourself.dart]:
```suggestion
        showShareBottomBar: false,
```

File: client/flutter/lib/components/emojiHeader.dart (hunk 1 above)
Comment (britannio) [reply client/flutter/lib/components/emojiHeader.dart]: There's actually a linter rule suggesting `this` shouldn't be used
https://dart-lang.github.io/linter/lints/unnecessary_this.html

File: client/flutter/lib/components/page_scaffold/page_scaffold.dart (hunk 5 above)
Comment (theswerd) [reply client/flutter/lib/components/page_scaffold/page_scaffold.dart]: Done

**Comment by theswerd** [comment]
All requested changes made

File: client/flutter/lib/components/page_scaffold/page_scaffold.dart (hunk 5 above)
Comment (PieterAelse) [inline client/flutter/lib/components/page_scaffold/page_scaffold.dart]: @theswerd You removed the `context` from `PageHeader` (👍), but not from this file, `PageScaffold` 😛 ...

**Pull request merged**